renderer/canvas.py

```diff
@@ -134,6 +134,91 @@ def _draw_item_image(self, surface, item_name, obs, position):
 
         self._draw_image(surface, f"{item_image_name}", position + self.pix_square_size * x_scale_factor, self.pix_square_size * y_scale_factor)
 
+    def _choose_container_asset(self, container_image_name, obs):
+        """
+        Helper function to choose the container asset based on the current 
+        true predicates in the state.
+
+        Meals can only be in containers, and cannot be drawn on their own. In the
+        state, the predicate "in" determines whether a meal is in a container.
+        Only one meal can be in each container at any point in time, so this 
+        helper function chooses the container asset based on the current
+        state of the meal. 
+
+        Args:
+            container_image_name (str): Name of the container
+            obs (List[Literal]): Game state predicates
+
+        Returns:
+            chosen_asset (str): Name of the chosen asset
+        """
+        container_image_name, container_id = trim_item_ID(container_image_name)
+        container_config = self.config["container"]["entities"][container_image_name]
+
+        # Get the name of the meal in the container
+        meal_name = None
+        for literal, is_true in obs.predicates.items():
+            if is_true and literal.name == "in" and literal.params[1].name == container_image_name + container_id:
+                meal_name = literal.params[0].name
+                break
+        
+        # If there is no meal in the container, use the default asset
+        if meal_name is None:
+            return container_config["assets"]["default"]
+        
+        # If there is a meal in the container, choose the asset based on the meal
+        item_predicates = {}
+        for literal, is_true in obs.predicates.items():
+            if is_true:
+                literal_args = [param.name for param in literal.params]
+                if meal_name in literal_args:
+                    item_predicates[literal.name] = [param.name for param in literal.params]
+        if len(item_predicates) == 0:
+            return container_config["assets"]["default"]
```

Comment:
i don't think this is ever hit so remove it

Reply:
removed